### PR TITLE
refactor: python reconstruction trajectory writing

### PR DESCRIPTION
**Pull request by andiwand**
rearrange writing mechanism for trajectories and reuse it for ambires and others in the future

### CODE REVIEW

**Comment by paulgessinger** [comment]
I'll have to touch this again when `Track` is done, but let's go ahead with this in any case.

**Comment by andiwand** [comment]
> I'll have to touch this again when `Track` is done, but let's go ahead with this in any case.

I feel like I am breaking the python interface on a weekly basis right now but I don't really see a way around this

File: Examples/Python/python/acts/examples/reconstruction.py [hunk 1]
```diff
@@ -129,6 +129,24 @@
...
+    defaults=[None] * 3,
+)
+
+AmbiguityResolutionConfig = namedtuple(
+    "AmbiguityResolutionConfig",
+    ["maximumSharedHits"],
+    defaults=[None] * 1,
+)
+
+
```
Comment (andiwand) [inline Examples/Python/python/acts/examples/reconstruction.py:144]:
```suggestion

```

File: Examples/Python/python/acts/examples/reconstruction.py [hunk 2]
```diff
@@ -577,6 +595,9 @@ def addSeeding(
     return s
 
 
+@acts.examples.NamedTypeArgs(
+    ckfPerformanceConfig=CKFPerformanceConfig,
+)
```
Comment (andiwand) [inline Examples/Python/python/acts/examples/reconstruction.py:600]:
```suggestion
```

**Review by benjaminhuth — APPROVED** [review]
Sorry I think I forgot this somehow, but I let's get this in now.

**Pull request merged**